### PR TITLE
Admin users and organizations

### PR DESCRIPTION
**Pull request by garrettqmartin8**
## Description

We are adding an `admin_users` table in order to mirror the way we have `office_users` and `tsp_users` set up. This will allow us to use login.gov more easily in staging, experimental, and dev. It will also allow us to separate admin authorization from the main `users` table.

## Setup

1. Update one of the existing users in your dev database to have `is_superuser` toggled to `true`. 
2. `make db_dev_migrate`
3.  You should now have an admin user record that matches the user record that you made into a superuser.
4. Ensure you also have an organizations table.

## Code Review Verification Steps

* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/167018034) for this change

### CODE REVIEW

File: migrations/20190702150158_create_admin_users.up.fizz [hunk 1]
```diff
@@ -0,0 +1,21 @@
+sql("CREATE TYPE admin_role AS ENUM ('SYSTEM_ADMIN', 'PROGRAM_ADMIN');")
```
Comment (Ryan-Koch) [inline migrations/20190702150158_create_admin_users.up.fizz]: yay enums!

File: pkg/models/admin_user.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	"github.com/gobuffalo/validate/validators"
+	"github.com/gofrs/uuid"
+)
+
+type AdminUser struct {
+	ID             uuid.UUID `json:"id" db:"id"`
+	CreatedAt      time.Time `json:"created_at" db:"created_at"`
+	UpdatedAt      time.Time `json:"updated_at" db:"updated_at"`
+	UserID         uuid.UUID `json:"user_id" db:"user_id"`
+	Role           string    `json:"role" db:"role"`
```
Comment (Ryan-Koch) [inline pkg/models/admin_user.go]: Would there be value in setting up a string type and const set up for the enum values (e.g. [shipment status](https://github.com/transcom/mymove/blob/master/pkg/models/shipment.go#L19))?

Comment (garrettqmartin8) [reply pkg/models/admin_user.go]: I don't see anything wrong with doing that, but I'm not super familiar with the convention. I would think about using a custom type if there was some extra behavior that was needed from that data, but I'm not sure that's the right way to look at it.

Comment (Ryan-Koch) [reply pkg/models/admin_user.go]: One thing I think we'd get in this case is enforcing the enum via the type in the go code that winds up using this thing. I could see that being somewhat useful, though I'll leave it in your hands about whether that's desirable enough to put into place here.

**Comment by garrettqmartin8** [comment]
@Ryan-Koch I also added `email` and `disabled` columns. The `email` will allow us to create a system admin user record and then have that person use login.gov. The `disabled` column mirrors the office users and tsp users.

File: pkg/models/admin_user.go (hunk 2 above)
Comment (garrettqmartin8) [reply pkg/models/admin_user.go]: I added a custom type for the admin and defined a method to enumerate the valid roles for the Pop validation. Let me know if the setup is in line with what you were thinking!

**Comment by Ryan-Koch** [comment]
> @Ryan-Koch I also added `email` and `disabled` columns. The `email` will allow us to create a system admin user record and then have that person use login.gov. The `disabled` column mirrors the office users and tsp users.

Most recent changes look good to me!

**Pull request merged**